lib/devise/failure_app.rb

```diff
@@ -142,11 +142,16 @@ def scope_url
 
       opts[:format] = request_format unless skip_format?
 
-      opts[:script_name] = relative_url_root if relative_url_root?
-
       router_name = Devise.mappings[scope].router_name || Devise.available_router_name
       context = send(router_name)
 
+      if relative_url_root?
+        opts[:script_name] = relative_url_root
+      elsif defined? context.routes
```

Comment:
We could create a private method to ask if the `root_path` is defined.

```ruby
module Devise
  class FailureApp < ActionController::Metal
    ...

    private

    def root_path_defined?(context)
      defined?(context.routes) && context.routes.url_helpers.root_path.present?
    end
  end
end
```

Wdyt?